RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -0,0 +1,318 @@
+  // can be deallocated from other, and vice versa.
+  // Always returns true for stateless allocators.
+  bool operator==(const AlignmentAllocator<T,N>& other) const {
+    return true;
+  }
+};
+
+
+class DAClusterizerInZT_vect  final : public TrackClusterizerInZ {
+
+public:
+  template<typename T> 
+  using AlignedVector = std::vector<T, AlignmentAllocator<T, 16> >;
+
+  // Internal data structure to 
+  struct track_t {
+    
+    void AddItem( double new_z, double new_t, double new_dz2, double new_dt2, const reco::TransientTrack* new_tt, double new_pi   )
+    {
+      z.push_back( new_z );
+      t.push_back( new_t );
+      dz2.push_back( new_dz2 );
+      dt2.push_back( new_dt2 );
+      errsum.push_back( 1./(1./new_dz2 + 1./new_dt2) );
+      tt.push_back( new_tt );
+      
+      pi.push_back( new_pi ); // track weight
+      Z_sum.push_back( 1.0 ); // Z[i]   for DA clustering, initial value as done in ::fill
+    }
+
+    
+    
+    unsigned int GetSize() const
+    {
+      return z.size();
+    }
+    
+    
+    // has to be called everytime the items are modified
+    void ExtractRaw()
+    {
+      _z = &z.front();
+      _t = &t.front();
+      _dz2 = &dz2.front();
+      _dt2 = &dt2.front();
+      _errsum = &errsum.front();
+      _Z_sum = &Z_sum.front();
+      _pi = &pi.front();
+    }
+    
+    double * __restrict__ _z __attribute__ ((aligned (16))); // z-coordinate at point of closest approach to the beamline
+    double * __restrict__ _t __attribute__ ((aligned (16))); // t-coordinate at point of closest approach to the beamline
+    double * __restrict__ _dz2 __attribute__ ((aligned (16))); // square of the error of z(pca)
+    double * __restrict__ _dt2 __attribute__ ((aligned (16))); // square of the error of t(pca)
+    double * __restrict__ _errsum __attribute__ ((aligned (16))); // sum of squares of the pca errors
+    
+    double * __restrict__  _Z_sum __attribute__ ((aligned (16))); // Z[i]   for DA clustering
+    double * __restrict__  _pi __attribute__ ((aligned (16))); // track weight
+    
+    AlignedVector<double> z; // z-coordinate at point of closest approach to the beamline
+    AlignedVector<double> t; // t-coordinate at point of closest approach to the beamline
+    AlignedVector<double> dz2; // square of the error of z(pca)
+    AlignedVector<double> dt2; // square of the error of t(pca)
+    AlignedVector<double> errsum; // sum of squares of the pca errors    
+    AlignedVector<double> Z_sum; // Z[i]   for DA clustering
+    AlignedVector<double> pi; // track weight
+
+    std::vector< const reco::TransientTrack* > tt; // a pointer to the Transient Track
+  };
+  
+  struct vertex_t {
+    AlignedVector<double> z; //           z coordinate
+    AlignedVector<double> t; //           t coordinate
+    AlignedVector<double> pk; //           vertex weight for "constrained" clustering
+    
+    // --- temporary numbers, used during update
+    AlignedVector<double> ei_cache;
```

Comment:
Can you add some comments about what these variables are, since the names are rather terse?